boards: dts: soc: bflb: misc fixes

kartben opened the pull request:
Use folders named "bflb" as it's the official vendor prefix.

Fixes BL604E not properly being listed under Bouffalo vnedor in board catalog (https://docs.zephyrproject.org/latest/boards/index.html#name=BL604E)

Also add `.. zephyr:board-supported-runners::` to the BL604E doc to automatically document supported runners.

kartben committed:
boards: bflb: remove redundant file reference

Board documentation page already has a link to the board's folder in
GitHub so hardcoding a link to the board's defconfig is redundant.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
boards: dts: soc: bflb: use proper folder names

Folders should be named after the vendor prefix

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: bflb: add .. zephyr:board-supported-runners

Add .. zephyr:board-supported-runners:: directive to document the
supported runners for the BL604E IOT DVK board.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

nandojve commented:
Nak on this.

You should not depends on vendor list defined for devicetree bindings to proper display the boards information. The information of the vendor can be extracted from the already defined field. https://github.com/zephyrproject-rtos/zephyr/blob/bf104d42cb26e34b03e3b9c8dbfa9dc6f4eecc85/boards/bouffalolab/bl60x/bl604e_iot_dvk/board.yml#L4

kartben commented:
> Nak on this.
> 
> You should not depends on vendor list defined for devicetree bindings to proper display the boards information. The information of the vendor can be extracted from the already defined field.
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/bf104d42cb26e34b03e3b9c8dbfa9dc6f4eecc85/boards/bouffalolab/bl60x/bl604e_iot_dvk/board.yml#L4

Sure, I can fix this (it turns out that many boards actually have wrong vendor in the board.yml, so it was simply easier to rely on folder name as this one is always right).

This doesn't change the fact that "bouffalolab" would be the only vendor in-tree not using the vendor prefix for its folder names which I am sure you will admit is probably not ideal?

kartben commented:
> You should not depends on vendor list defined for devicetree bindings to proper display the boards information. 

See #89024.
But folder names should be fixed nevertheless.

VynDragon reviewed (dismissed):
Makes sense to me.

nandojve commented:
> This doesn't change the fact that "bouffalolab" would be the only vendor in-tree not using the vendor prefix for its folder names which I am sure you will admit is probably not ideal?

IMHO this should not be an excuse to change the folder name. Right now the tool seems to relying on the folder name which I believe it is not be the best way. On my perspective, the new hardware model introduced freedom to allow us to have different folder structures, define metadata (both SoC and Board) and expand. The metadata binds the folder structure defined by the vendor and that should be the ideal source of information. This way we can have freedom on the folder structure names to make it easy to ready and organize the content.

So, if we enforce people to use the metadata and fill any the missing fields the tools can collect the correct information in the right way from the right place. We can share the code that collect the info and use that as standard instead "force" naming and avoid multiple source of truth.

kartben commented:
> > This doesn't change the fact that "bouffalolab" would be the only vendor in-tree not using the vendor prefix for its folder names which I am sure you will admit is probably not ideal?
> 
> IMHO this should not be an excuse to change the folder name. Right now the tool seems to relying on the folder name which I believe it is not be the best way. On my perspective, the new hardware model introduced freedom to allow us to have different folder structures, define metadata (both SoC and Board) and expand. The metadata binds the folder structure defined by the vendor and that should be the ideal source of information. This way we can have freedom on the folder structure names to make it easy to ready and organize the content.
> 
> So, if we enforce people to use the metadata and fill any the missing fields the tools can collect the correct information in the right way from the right place. We can share the code that collect the info and use that as standard instead "force" naming and avoid multiple source of truth.

Yeah, but it's just the way boards and socs [must be organized](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#create-your-board-directory:~:text=If%20the%20vendor%20is%20defined%20in%20the%20list%20in%20dts/bindings/vendor%2Dprefixes.txt%20then%20you%20must%20use%20that%20vendor%20prefix%20as%20%3CVENDOR%3E.), sorry.

If you really want, I can drop the change to `dts/riscv/bouffalolab` since that one is not actually a requirement as far as I can tell.

nandojve commented:
> Yeah, but it's just the way boards and socs [must be organized].

I see, thank you for pointing out.
Go ahead and keep same folders for dts.

Note: Not sure if it is necessary change this line too
https://github.com/zephyrproject-rtos/zephyr/blob/b469fc188b4fc87c64e598618c07d717bc570c9d/soc/bouffalolab/soc.yml#L2

kartben commented:
> > Yeah, but it's just the way boards and socs [must be organized].
> 
> I see, thank you for pointing out. Go ahead and keep same folders for dts.
> 
> Note: Not sure if it is necessary change this line too
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/b469fc188b4fc87c64e598618c07d717bc570c9d/soc/bouffalolab/soc.yml#L2

Yeah good point.
Maybe @nordicjm can comment here but my recommendation would be to actually just remove the family name altogether. AFAIK it's not mandatory to have a family (check e.g. all socs with "<no family>" in the [board catalog](https://docs.zephyrproject.org/latest/boards/index.html#)), and in this case would be better than making up a name for something that doesn't really correspond to anything in the vendor catalog (as opposed to, say, Atmel SAM0 family)

nordicjm commented:
> > > Yeah, but it's just the way boards and socs [must be organized].
> > 
> > 
> > I see, thank you for pointing out. Go ahead and keep same folders for dts.
> > Note: Not sure if it is necessary change this line too
> > https://github.com/zephyrproject-rtos/zephyr/blob/b469fc188b4fc87c64e598618c07d717bc570c9d/soc/bouffalolab/soc.yml#L2
> 
> Yeah good point. Maybe @nordicjm can comment here but my recommendation would be to actually just remove the family name altogether. AFAIK it's not mandatory to have a family (check e.g. all socs with "<no family>" in the [board catalog](https://docs.zephyrproject.org/latest/boards/index.html#)), and in this case would be better than making up a name for something that doesn't really correspond to anything in the vendor catalog (as opposed to, say, Atmel SAM0 family)

Family name is completely optional, it can go, or be renamed

VynDragon commented:
Ah so if it can go i'd like to take this opportunity to recommend making it either go back (? maybe not, different code trees) to SOC_FAMILY_BFLB instead of SOC_FAMILY_BOUFFALOLAB_BFLB, or consider that BL60x is the family (bl602 family in bflb terms) and the socs are really bl602 and bl604, matched to footprints. I am not sure what the best way would be though.

kartben commented:
@nandojve do you want me to add the dropping of the family name to this PR?

VynDragon commented:
> @nandojve do you want me to add the dropping of the family name to this PR?
> 
> EDIT: posted at the same time as @VynDragon, woops. I'm fine with SOC_FAMILY_BFLB, but having no family and just working at the "series" level would be fine too IMO (e.g CMakeLists.txt for the hal would test for SOC_SERIES_BL60X instead of SOC_FAMILY_BOUFFALOLAB_BFLB, etc.). Will let @nandojve make the call :)

All the rest of the code is based on SOC_SERIES_BLxxx checks anyway

kartben committed:
soc: bflb: rename bouffalolab_bflb soc family to bflb

For simplicity/consistency with many other soc families, rename the
bouffalolab_bflb soc family to the simpler bflb.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

VynDragon commented on soc/bflb/bl60x/Kconfig.soc:
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2021-2025 ATL Electronics
```
Is it normal this file is marked as added? Did the previous file not get removed somehow?

VynDragon replied:
Yea i checked the the original file makes a return in the second commit.

kartben replied:
Thanks for paying attention! Still trying to get my head around how that happened, but should be good now, please check again
